diesel_derives/src/lib.rs

```diff
@@ -597,6 +597,154 @@ pub fn derive_sql_type(input: TokenStream) -> TokenStream {
 /// #     Ok(())
 /// # }
 /// ```
+///
+/// ## Custom Aggregate Functions
+/// Custom aggregate functions can be created in SQLite by adding the `aggregate`
+/// attribute and implementing the
+/// [SqliteAggregateFunction](../diesel/sqlite/trait.SqliteAggregateFunction.html) trait.
+///
+/// ```rust
+/// # extern crate diesel;
+/// # use diesel::*;
+/// #
+/// # #[cfg(feature = "sqlite")]
+/// # fn main() {
+/// #   run().unwrap();
+/// # }
+/// #
+/// # #[cfg(not(feature = "sqlite"))]
+/// # fn main() {
+/// # }
+/// use diesel::sql_types::Integer;
+/// use diesel::sqlite::SqliteAggregateFunction;
+/// use std::ops::AddAssign;
+///
+/// sql_function! {
+///     #[aggregate]
+///     fn my_sum(x: Integer) -> Integer;
+/// }
+///
+/// #[derive(Default)]
+/// struct MySum<T> { sum: T }
+///
+/// impl<T: Default + Copy + AddAssign> SqliteAggregateFunction<T> for MySum<T> {
+///     type Output = T;
+///
+///     fn step(&mut self, expr: T) {
+///         self.sum += expr;
+///     }
+///
+///     fn finalize(&self) -> Self::Output {
+///         self.sum
+///     }
+/// }
+/// # table! {
+/// #     players {
+/// #         id -> Integer,
+/// #         score -> Integer,
+/// #     }
+/// # }
+///
+/// # #[cfg(feature = "sqlite")]
+/// fn run() -> Result<(), Box<::std::error::Error>> {
+/// #    use self::players::dsl::*;
+///     let connection = SqliteConnection::establish(":memory:")?;
+/// #    connection.execute("create table players (id integer primary key autoincrement, score integer)").unwrap();
+/// #    connection.execute("insert into players (score) values (10), (20), (30)").unwrap();
+///
+///     my_sum::register_impl::<MySum<i32>, _>(&connection)?;
+///
+///     let result = players.select(my_sum(score))
+///         .get_result::<i32>(&connection)?;
+///
+///     println!("The total score of all the players is: {}", result);
+///
+/// #    assert_eq!(60, result);
+///     Ok(())
+/// }
+/// ```
+///
+/// With multiple parameters
```

Comment:
Some more details here would be great, so something like
```suggestion
/// With multiple function arguments the arguments are passed as tuple to `SqliteAggregateFunction`
```